relay-spans/src/span.rs

```diff
@@ -0,0 +1,350 @@
+use std::str::FromStr;
+
+use chrono::{TimeZone, Utc};
+use serde::Deserialize;
+use serde_repr::Deserialize_repr;
+
+use relay_event_schema::protocol::{Span as EventSpan, SpanId, SpanStatus, Timestamp, TraceId};
+use relay_protocol::{Annotated, Object, Value};
+
+use crate::status_codes;
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Span {
+    pub trace_id: String,
+    pub span_id: String,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub trace_state: String,
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    pub parent_span_id: Option<String>,
+    pub name: String,
+    pub kind: SpanKind,
+    pub start_time_unix_nano: i64,
+    pub end_time_unix_nano: i64,
+    pub attributes: Vec<KeyValue>,
+    pub dropped_attributes_count: u32,
+    pub events: Vec<Event>,
+    pub dropped_events_count: u32,
+    pub links: Vec<Link>,
+    pub dropped_links_count: u32,
+    #[serde(default)]
+    pub status: Status,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Event {
+    pub time_unix_nano: u64,
+    pub name: String,
+    pub attributes: Vec<KeyValue>,
+    pub dropped_attributes_count: u32,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Link {
+    pub trace_id: String,
+    pub span_id: String,
+    pub trace_state: String,
```

Comment:
Same question for all the fields on all the sub-structs: Is this field optional? If so, `#[serde(default)]`.